kythe/rust/indexer/src/error.rs

```diff
@@ -0,0 +1,27 @@
+// Copyright 2020 The Kythe Authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/// An enum containing all the errors that this library can return.
+pub enum KytheError {
```

Comment:
```suggestion
#[derive(Debug, Clone, Eq, PartialEq)]
pub enum KytheError {
```

The `Error` and `Display` traits should also be implemented by hand.

Reply:
Derived debug, but Clone, Eq, and PartialEq can't be derived in the new commit due to the stored errors not deriving these traits

Reply:
That's fine; `Debug`, `Display`, and `Error` are by far the important ones.